compiler/rustc_passes/src/dead.rs

```diff
@@ -958,18 +998,36 @@ fn check_mod_deathness(tcx: TyCtxt<'_>, module: LocalModDefId) {
     let module_items = tcx.hir_module_items(module);
 
     for item in module_items.items() {
-        if let hir::ItemKind::Impl(impl_item) = tcx.hir().item(item).kind {
-            let mut dead_items = Vec::new();
-            for item in impl_item.items {
-                let def_id = item.id.owner_id.def_id;
-                if !visitor.is_live_code(def_id) {
+        let dead_codes = {
+            let filter_dead_item = |def_id| {
```

Comment:
The closure can be bound earlier, there is no need to create a scope just for it.